### PR TITLE
add update, findAdll and findById methods for notification template

### PR DESCRIPTION
**Pull request by DimaSannytski**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

**Review by Max877866 — APPROVED** [review]

**Review by Natalia1910 — APPROVED** [review]

**Review by KuzbytMaksym — APPROVED** [review]

File: dao/src/main/java/greencity/repository/NotificationParameterRepository.java [hunk 1]
```diff
@@ -4,6 +4,12 @@
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.stereotype.Repository;
 
+import java.util.List;
+
 @Repository
 public interface NotificationParameterRepository extends JpaRepository<NotificationParameter, Long> {
+    /**
+     * {@inheritDoc}
+     */
```
Comment (VadymMakitra) [inline dao/src/main/java/greencity/repository/NotificationParameterRepository.java:13]: Don't see any reason for override findAll, JpaRepository have default methods findAll();

File: service-api/src/main/java/greencity/constant/ErrorMessage.java [hunk 2]
```diff
@@ -89,6 +89,7 @@
     public static final String SOME_CERTIFICATES_ARE_INVALID =
         "SOME CERTIFICATES ARE INVALID. A valid certificate is listed here";
     public static final String CERTIFICATE_NOT_FOUND = "CERTIFICATE_NOT_FOUND";
+    public static final String NOTIFICATION_TEMPLATE_NOT_FOUND = "couldn't found notification template with id:";
```
Comment (VadymMakitra) [inline service-api/src/main/java/greencity/constant/ErrorMessage.java:92]: Will be better to write here: Notification template with id doesn't exist;

File: service/src/main/java/greencity/service/notification/ManagementNotificationServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+     * {@inheritDoc}
+     */
+    @Override
+    public NotificationTemplateDto findById(Long id) {
+        return modelMapper.map(getById(id), NotificationTemplateDto.class);
+    }
+
+    private NotificationTemplate getById(Long id) {
+        return notificationTemplateRepository.findNotificationTemplateById(id)
+            .orElseThrow(() -> new NotFoundException(ErrorMessage.NOTIFICATION_TEMPLATE_NOT_FOUND + id));
```
Comment (VadymMakitra) [inline service/src/main/java/greencity/service/notification/ManagementNotificationServiceImpl.java:66]: Please add a test for cover this exception;

File: service/src/test/java/greencity/service/notification/ManagementNotificationServiceImplTest.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    @Mock
+    NotificationTemplateRepository templateRepository;
+    @Mock
+    ModelMapper modelMapper;
+
+    @InjectMocks
+    private ManagementNotificationServiceImpl notificationService;
+
+    @Test
+    void findAll() {
```
Comment (VadymMakitra) [inline service/src/test/java/greencity/service/notification/ManagementNotificationServiceImplTest.java:30]: Please add AssertEquals for make sure that you get correct object.

File: service/src/main/java/greencity/service/notification/ManagementNotificationServiceImpl.java (hunk 3 above)
Comment (VadymMakitra) [inline service/src/main/java/greencity/service/notification/ManagementNotificationServiceImpl.java:66]: Please make sure you have Controller test that verifies case with BadRequest.

**Review by VadymMakitra — CHANGES_REQUESTED** [review]

File: dao/src/main/java/greencity/repository/NotificationParameterRepository.java [hunk 5]
```diff
@@ -4,6 +4,12 @@
...
+    List<NotificationParameter> findAll();
```
Comment (VadymMakitra) [inline dao/src/main/java/greencity/repository/NotificationParameterRepository.java:14]: Don't see any reason for override findAll, JpaRepository have default methods findAll();

**Review by VadymMakitra — CHANGES_REQUESTED** [review]

File: core/src/main/java/greencity/controller/ManagementNotificationController.java [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+@RestController
+@RequestMapping("/admin/notification")
+@Validated
+public class ManagementNotificationController {
+    private final ManagementNotificationService notificationService;
+
+    /**
+     * Constructor with parameters.
+     */
+    @Autowired
```
Comment (thesvider) [inline core/src/main/java/greencity/controller/ManagementNotificationController.java:28]: you don't need to declare constructor explicitly. Use @ RequiredArgsConstrutor lombok annotation instead

File: core/src/main/java/greencity/controller/ManagementNotificationController.java [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+import springfox.documentation.annotations.ApiIgnore;
+
+@RestController
+@RequestMapping("/admin/notification")
+@Validated
```
Comment (thesvider) [inline core/src/main/java/greencity/controller/ManagementNotificationController.java:21]: You have annotation @ Validated but none of the controller method parameters are marked as Valid. Does validation work for any of the endpoints here?

File: dao/src/main/java/greencity/entity/notifications/NotificationTemplate.java [hunk 8]
```diff
@@ -15,6 +12,7 @@
 @AllArgsConstructor
 @EqualsAndHashCode
 @NoArgsConstructor
+@Builder
```
Comment (thesvider) [inline dao/src/main/java/greencity/entity/notifications/NotificationTemplate.java:15]: why do you need both @ Data and @ Builder implementations?  The both provide the way to build your object: first - using chain of setters and second - with builder methods.

File: dao/src/main/java/greencity/repository/NotificationParameterRepository.java (hunk 5 above)
Comment (thesvider) [reply dao/src/main/java/greencity/repository/NotificationParameterRepository.java:14]: @DimaSannytski , please comment on this

File: service-api/src/main/java/greencity/dto/NotificationTemplateDto.java [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
```
Comment (thesvider) [inline service-api/src/main/java/greencity/dto/NotificationTemplateDto.java:11]: don't use both Builder and Data

File: service-api/src/main/java/greencity/service/notification/ManagementNotificationService.java [hunk 10]
```diff
@@ -0,0 +1,22 @@
+package greencity.service.notification;
+
+import greencity.dto.NotificationTemplateDto;
+import greencity.dto.PageableDto;
+import org.springframework.data.domain.Pageable;
+
+public interface ManagementNotificationService {
+    /**
```
Comment (thesvider) [inline service-api/src/main/java/greencity/service/notification/ManagementNotificationService.java:8]: what is the purpose of specifying @ inheritDoc everywhere? these methods are custom repo methods. They won't inherit any documentation. right?

**Review by thesvider — CHANGES_REQUESTED** [review]
please resolve comments

File: service-api/src/main/java/greencity/service/notification/ManagementNotificationService.java (hunk 10 above)
Comment (DimaSannytski) [reply service-api/src/main/java/greencity/service/notification/ManagementNotificationService.java:8]: mvn checkstyle:check gives an error if there is no documentation

File: core/src/main/java/greencity/controller/ManagementNotificationController.java [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+     */
+    @ApiOperation("Update notification template")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = HttpStatuses.OK),
+        @ApiResponse(code = 400, message = HttpStatuses.BAD_REQUEST),
+        @ApiResponse(code = 401, message = HttpStatuses.UNAUTHORIZED),
+        @ApiResponse(code = 403, message = HttpStatuses.FORBIDDEN),
+        @ApiResponse(code = 404, message = HttpStatuses.NOT_FOUND)
+    })
+    @PostMapping(value = "/update")
```
Comment (thesvider) [inline core/src/main/java/greencity/controller/ManagementNotificationController.java:55]: PUT method is more suitable for update. Also, according to restful spec you should keep @ PostMapping empty

File: service-api/src/main/java/greencity/service/notification/ManagementNotificationService.java (hunk 10 above)
Comment (thesvider) [reply service-api/src/main/java/greencity/service/notification/ManagementNotificationService.java:8]: please add meaningful javadoc instead.

**Review by VadymMakitra — APPROVED** [review]

**Review by thesvider — APPROVED** [review]

**Pull request merged**